src/org/omegat/gui/editor/EditorUtils.java

```diff
@@ -99,6 +101,47 @@ public static int getWordEnd(JTextComponent c, int offs) throws BadLocationExcep
         return result;
     }
 
+    private static int getWordBoundary(JTextComponent c, int offs, boolean end) throws BadLocationException {
+        int result = offs;
+        Element line = Utilities.getParagraphElement(c, offs);
+        if (line == null) {
+            throw new BadLocationException("No word at " + offs, offs);
+        }
+        int lineStart = line.getStartOffset();
+        Document doc = c.getDocument();
+        int lineEnd = Math.min(line.getEndOffset(), doc.getLength());
+        if  (lineEnd - lineStart > 0) {
+            String lineString = doc.getText(lineStart, lineEnd - lineStart);
+            // Detection of target string locale.
+            // A default is UI component locale, and when OmegaT project
+            // is loaded, it uses a source or a target language as a
+            // processing locale.
+            Locale locale = c.getLocale();
+            if (c instanceof EditorTextArea3 && Core.getProject().isProjectLoaded()) {
```

Comment:
It is better to pass boolean `isTranslation` by caller because we need to detect caller component here, and use `(EditorTextArea3) c.isActivaTranslation(offs)` and the `offs` is passed by caller.